Lib/test/test_capi/test_opt.py

```diff
@@ -1940,6 +1940,23 @@ def testfunc(n):
         self.assertNotIn("_COMPARE_OP_INT", uops)
         self.assertNotIn("_GUARD_IS_TRUE_POP", uops)
 
+    def test_call_isinstance_guards_removed(self):
+        def testfunc(n):
+            x = 0
+            for _ in range(n):
+                y = isinstance(42, int)
+                if y:
+                    x += 1
+            return x
+
+        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
+        self.assertEqual(res, TIER2_THRESHOLD)
+        self.assertIsNotNone(ex)
+        uops = get_opnames(ex)
+        self.assertIn("_CALL_ISINSTANCE", uops)
+        self.assertNotIn("_GUARD_CALLABLE_ISINSTANCE_NULL", uops)
```

Comment:
```suggestion
        self.assertNotIn("_GUARD_THIRD_NULL", uops)
```

Reply:
oops, thanks for catching this! One thing I noticed after having written a couple of these tests is that it's quite easy to make a mistake like this (as evidenced here :D) which might go unnoticed since `assertNotIn` will pretty much always pass..

I think the tests could be more robust if there was a way to run the executors without optimizing the uops. Each test could then first use `assertIn` to ensure the relevant uops are in fact present and only then run with optimizations enabled. Though dynamically turning off optimizations would require some changes in the interpreter so I'm not sure if it's worth it.

Reply:
I think Mark wanted to do something what you're suggesting at one point.